server/sonar-server/src/main/java/org/sonar/server/computation/taskprocessor/CeProcessingSchedulerImpl.java

```diff
@@ -39,16 +39,17 @@
   private static final Logger LOG = Loggers.get(CeProcessingSchedulerImpl.class);
 
   private final CeProcessingSchedulerExecutorService executorService;
-  private final CeWorkerCallable workerRunnable;
+  private final CeWorker workerRunnable;
 
   private final long delayBetweenTasks;
   private final TimeUnit timeUnit;
   private final ChainingCallback[] chainingCallbacks;
 
   public CeProcessingSchedulerImpl(CeConfiguration ceConfiguration,
-    CeProcessingSchedulerExecutorService processingExecutorService, CeWorkerCallable workerRunnable) {
+                                   CeProcessingSchedulerExecutorService processingExecutorService,
+                                   CeWorkerFactory ceWorkerFactory) {
     this.executorService = processingExecutorService;
-    this.workerRunnable = workerRunnable;
+    this.workerRunnable = ceWorkerFactory.create();
```

Comment:
(adding my comment edit here to trigger notification)

the factory should be called at this line https://github.com/SonarSource/sonarqube/blob/master/server/sonar-server/src/main/java/org/sonar/server/computation/taskprocessor/CeProcessingSchedulerImpl.java#L59 as parameter of the `ChainingCallback` class which should now hold a `CeWorker` property rather than use the one of the `CeProcessingSchedulerImpl`